woocommerce/google-listings-and-ads
# Only submit 'Published' products

### Changes proposed in this Pull Request:

This PR adds a check to the product syncer and repository to only submit "Published" products to the Merchant Center and avoid submitting "Draft" or "Pending Review" ones.

It also deletes the product from Merchant Center if its status is changed to anything other than "Published".

Note: We might want to handle the case where products' visibility is set to `private` or `password protected` too. I'm not sure if these types of products should be submitted to MC because their data would not be publicly available in the shop.

### Detailed test instructions:

1. Create a draft product and save it
2. Make sure it is not synced to MC
3. Edit the product and publish it
4. It should now sync to MC
5. Edit the product again but now change the status to "Pending Review"
6. The synced product should be deleted from MC

### Changelog Note:

> Fix - Only sync published products
